main/src/ui/conversation_content_view/conversation_item_skeleton.vala

```diff
@@ -123,6 +123,11 @@ public class ItemMetaDataHeader : Box {
 
         update_name_label();
         name_label.style_updated.connect(update_name_label);
+        stream_interactor.get_module(RosterManager.IDENTITY).updated_roster_item.connect((account, jid, roster_item) => {
```

Comment:
```suggestion
        updated_roster_handler_id = stream_interactor.get_module(RosterManager.IDENTITY).updated_roster_item.connect((account, jid, roster_item) => {
```

You connect a listener to the `RosterManager`, but you're never detaching it. That means the code can be called even after the `ConversationItemSkeleton` has been destroyed.
Connecting the listener will return a handler id, which you can use to detach it afterwards. The needed code is similar to how it's done with the `time_update_timeout` in the same file.

Reply:
I added that in the new commit. 

Small side not:
In this case (with conversation replaced with this.conversation) you don't need too disconnect a signal. In the generated c code you get something like  
```
_tmp8_ = g_signal_connect_object (_tmp7_, "updated-roster-item", (GCallback) ___lambda4__dino_roster_manager_updated_roster_item, self, 0);
self->priv->updated_roster_handler_id = _tmp8_;
```
but g_signal_connect_object disconnects signal automatically but such behavior is not documented, possible a bug and easy to break.